templates/glossary.md

```diff
@@ -202,6 +202,21 @@ A heavy `rfl` occurs when `rfl` is asked to perform many steps of definitional r
 A collection of successively more constrained [typeclasses](#class) within a related area of mathematics.
 In [mathlib](#mathlib), we have the *algebraic hierarchy* (`semiring`, `ring`, `field`, ...), the *order hierarchy* (`preorder`, `partial_order`, `linear_order`, ...), the *topological hierarchy* (`t1_space`, `t2_space`, `normal_space`, ...), the *categorical hierarchy* (`preadditive`, `abelian`, `monoidal`, ...) but also the *scalar hierarchy* (`mul_action`, `distrib_mul_action`, `module`, ...), the *norm hierarchy*, and intersections of previous ones like the *order-algebraic hierarchy*, the *topologico-algebraic hierarchy*, and others.
 
+### HoTT
+
+*Homotopy Type Theory*, a [type theory](#type-theory) distinguished by the inclusion of an additional *univalence axiom*, which makes precise the notion that two disparate implementations of a type which would be thought of as equivalent are also in a mathematical sense equal.
+
+In Lean 2, there was native support within [core Lean](#core-lean) itself for a side-by-side [homotopy type theory-based library](https://github.com/leanprover/lean2/blob/8072fdf9a0b31abb9d43ab894d7a858639e20ed7/hott/hott.md).
+
+Lean 3's [kernel](kernel) introduced singleton elimination, which is inconsistent with HoTT's univalence axiom.
+As a result, core Lean 3 no longer shipped a HoTT library.
+A partial port of the Lean 2 HoTT library for Lean 3 was made in [an external project by Gabriel Ebner and fellow contributors](https://github.com/gebner/hott3).
+
+As an illustrative concrete example, there are many [definitions of the type of natural numbers](https://en.wikipedia.org/wiki/Natural_number#Formal_definitions) which can be viewed as constructing equivalent mathematical objects.
+Core Lean has a [Peano-esque implementation](https://leanprover-community.github.io/mathlib_docs/init/core.html#nat) of them, and [mathlib](#mathlib) has an additional [binary representation-based one](https://leanprover-community.github.io/mathlib_docs/data/num/basic.html#pos_num) which is shown to be equivalent.
+Given that Lean does *not* have the univalence axiom, these two types are equivalent, but they are not *equal* as types.
```

Comment:
```suggestion
Given that Lean does *not* have the univalence axiom, these two types are equivalent, but they are not provably *equal* as types.
```
It's consistent in Lean 3, but not provable, that any two types of the same cardinality are equal.